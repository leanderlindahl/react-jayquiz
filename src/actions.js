export const SET_CURRENT_ANSWER = 'SET_CURRENT_ANSWER';

export const SET_CURRENT_ANSWER_STATUS = 'SET_CURRENT_ANSWER_STATUS';

export const SET_DISPLAY_ANSWER_RESPONSE = 'SET_DISPLAY_ANSWER_RESPONSE';

export const SET_SCORE = 'SET_SCORE';
