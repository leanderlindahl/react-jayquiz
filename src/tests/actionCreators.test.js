import {
  fetchQuestions,
  receiveQuestions,
  requestQuestions,
  setCurrentAnswerStatus,
  setCurrentQuestionIndex,
  setDisplayAnswerResponse,
  setGameOver,
  setOptionsDisabled,
  setOutOfTime,
  setQuestionNumber,
  setQuestionsPerRound,
  setSelectedOption,
  setScore,
  setTimedOutAnswers,
  setWrongAnswers,
  setUsedQuestions
} from '../actionCreators';
import { RECEIVE_QUESTIONS } from '../actions';

const jsonQuestionObject = {
  type: RECEIVE_QUESTIONS,
  category: 18,
  results: [
    {
      category: 'Politics',
      type: 'multiple',
      difficulty: 'easy',
      question: 'Whose 2016 presidential campaign slogan was &quot;Make America Great Again&quot;?',
      correct_answer: 'Donald Trump',
      incorrect_answers: ['Ted Cruz', 'Marco Rubio', 'Bernie Sanders']
    },
    {
      category: 'Entertainment: Music',
      type: 'multiple',
      difficulty: 'medium',
      question:
        'Which French duo had UK hits in 1998 with the songs &#039;Sexy Boy&#039;, &#039;Kelly Watch The Stars&#039; &amp; &#039;All I Need&#039;?',
      correct_answer: 'Air',
      incorrect_answers: ['Fire', 'Earth', 'Water']
    }
  ],
  receivedAt: 1541764671426
};

const jsonQuestionResponse = {
  type: 'RECEIVE_QUESTIONS',
  category: 18,
  questions: [
    {
      category: 'Politics',
      type: 'multiple',
      difficulty: 'easy',
      question: 'Whose 2016 presidential campaign slogan was &quot;Make America Great Again&quot;?',
      correct_answer: 'Donald Trump',
      incorrect_answers: ['Ted Cruz', 'Marco Rubio', 'Bernie Sanders']
    },
    {
      category: 'Entertainment: Music',
      type: 'multiple',
      difficulty: 'medium',
      question:
        'Which French duo had UK hits in 1998 with the songs &#039;Sexy Boy&#039;, &#039;Kelly Watch The Stars&#039; &amp; &#039;All I Need&#039;?',
      correct_answer: 'Air',
      incorrect_answers: ['Fire', 'Earth', 'Water']
    }
  ],
  receivedAt: 1530518207007
};

describe('actionCreators', () => {
  const literallyJustDateNow = () => Date.now();

  it('fetchQuestions', () => {
    expect(fetchQuestions(jsonQuestionObject.questions)).toMatchSnapshot();
  });

  it('requestQuestions', () => {
    expect(requestQuestions(18)).toMatchSnapshot();
  });

  it('receiveQuestions', () => {
    // https://codewithhugo.com/mocking-the-current-date-in-jest-tests/
    const realDateNow = Date.now.bind(global.Date);
    const dateNowStub = jest.fn(() => 1530518207007);
    global.Date.now = dateNowStub;
    expect(receiveQuestions(18, jsonQuestionObject)).toEqual(jsonQuestionResponse);
    global.Date.now = realDateNow;
  });

  it('setCurrentAnswerStatus', () => {
    expect(setCurrentAnswerStatus('right')).toMatchSnapshot();
  });

  it('setCurrentQuestionIndex', () => {
    expect(setCurrentQuestionIndex(2)).toMatchSnapshot();
  });

  it('setDisplayAnswerResponse', () => {
    expect(setDisplayAnswerResponse('true')).toMatchSnapshot();
  });

  it('setGameOver', () => {
    expect(setGameOver('false')).toMatchSnapshot();
  });

  it('setOptionsDisabled', () => {
    expect(setOptionsDisabled('true')).toMatchSnapshot();
  });

  it('setOutOfTime', () => {
    expect(setOutOfTime(true)).toMatchSnapshot();
  });

  it('setQuestionNumber', () => {
    expect(setQuestionNumber(2)).toMatchSnapshot();
  });

  it('setQuestionsPerRound', () => {
    expect(setQuestionsPerRound(10)).toMatchSnapshot();
  });

  it('setSelectedOption', () => {
    expect(setSelectedOption('8')).toMatchSnapshot();
  });

  it('setScore', () => {
    expect(setScore(2)).toMatchSnapshot();
  });

  it('setTimedOutAnswers', () => {
    expect(setTimedOutAnswers(2)).toMatchSnapshot();
  });

  it('setWrongAnswers', () => {
    expect(setWrongAnswers(2)).toMatchSnapshot();
  });

  it('setUsedQuestions', () => {
    expect(setUsedQuestions(['questions alfa', 'question beta'])).toMatchSnapshot();
  });
});
