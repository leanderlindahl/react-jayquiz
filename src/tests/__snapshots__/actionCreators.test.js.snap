// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`actionCreators fetchQuestions 1`] = `[Function]`;

exports[`actionCreators requestQuestions 1`] = `
Object {
  "category": 18,
  "type": "REQUEST_QUESTIONS",
}
`;

exports[`actionCreators setCurrentAnswerStatus 1`] = `
Object {
  "payload": "right",
  "type": "SET_CURRENT_ANSWER_STATUS",
}
`;

exports[`actionCreators setCurrentQuestionIndex 1`] = `
Object {
  "payload": 2,
  "type": "SET_CURRENT_QUESTION_INDEX",
}
`;

exports[`actionCreators setDisplayAnswerResponse 1`] = `
Object {
  "payload": "true",
  "type": "SET_DISPLAY_ANSWER_RESPONSE",
}
`;

exports[`actionCreators setGameOver 1`] = `
Object {
  "payload": "false",
  "type": "SET_GAME_OVER",
}
`;

exports[`actionCreators setOptionsDisabled 1`] = `
Object {
  "payload": "true",
  "type": "SET_OPTIONS_DISABLED",
}
`;

exports[`actionCreators setOutOfTime 1`] = `
Object {
  "payload": true,
  "type": "SET_OUT_OF_TIME",
}
`;

exports[`actionCreators setQuestionNumber 1`] = `
Object {
  "payload": 2,
  "type": "SET_QUESTION_NUMBER",
}
`;

exports[`actionCreators setQuestionsPerRound 1`] = `
Object {
  "payload": 10,
  "type": "SET_QUESTIONS_PER_ROUND",
}
`;

exports[`actionCreators setScore 1`] = `
Object {
  "payload": 2,
  "type": "SET_SCORE",
}
`;

exports[`actionCreators setSelectedOption 1`] = `
Object {
  "payload": "8",
  "type": "SET_SELECTED_OPTION",
}
`;

exports[`actionCreators setTimedOutAnswers 1`] = `
Object {
  "payload": 2,
  "type": "SET_TIMED_OUT_ANSWERS",
}
`;

exports[`actionCreators setUsedQuestions 1`] = `
Object {
  "payload": Array [
    "questions alfa",
    "question beta",
  ],
  "type": "SET_USED_QUESTIONS",
}
`;

exports[`actionCreators setWrongAnswers 1`] = `
Object {
  "payload": 2,
  "type": "SET_WRONG_ANSWERS",
}
`;
